nrf52-hal-common/src/uarte.rs

```diff
@@ -19,22 +25,37 @@ use crate::gpio::{
     PushPull,
 };
 
+use bbqueue::{BBQueue, Producer, Consumer, typenum::*, GrantR};
+
 // Re-export SVD variants to allow user to directly set values
 pub use crate::target::uarte0::{
     baudrate::BAUDRATEW as Baudrate,
     config::PARITYW as Parity,
 };
 
 pub trait UarteExt: Deref<Target = uarte0::RegisterBlock> + Sized {
-    fn constrain(self, pins: Pins, parity: Parity, baudrate: Baudrate) -> Uarte<Self>;
+    fn constrain(self, nvic: NVIC, pins: Pins, parity: Parity, baudrate: Baudrate) -> Uarte<Self>;
 }
 
 impl UarteExt for UARTE0 {
-    fn constrain(self, pins: Pins, parity: Parity, baudrate: Baudrate) -> Uarte<Self> {
-        Uarte::new(self, pins, parity, baudrate)
+    fn constrain(self, nvic: NVIC, pins: Pins, parity: Parity, baudrate: Baudrate) -> Uarte<Self> {
+        Uarte::new(self, nvic, pins, parity, baudrate)
     }
 }
 
+lazy_static! {
+    static ref MUH_BUFFAH: BBQueue<U2048> = {
+        BBQueue::new()
+    };
+}
```

Comment:
This is a single `static`, and this module only supports UARTE0 right now, so this works. But what if we add support for other UARTE instances?

Maybe we could make the user allocate the buffer for us? That would mean the API would need an additional lifetime, but the user can always fall back to allocating the buffer in a `static` (thereby making the lifetime `'static`), if that causes any trouble.

Reply:
Updated the interface to take a `Producer` and `Consumer`, which no longer require lifetimes (they explicitly require a 'static buffer for backing, but I have some helper methods now for that in v0.2.1).

Reply:
Also if we add support for other UARTE instances, I think those will need their own separate interrupt handler, where we can grab the "proper" `static Consumer`, and combine the logic. Something like

```rust
#[interrupt]
unsafe fn UARTE0_UART0() {
    let uart = &*UARTE0::ptr();
    let cons: &mut Consumer = MAYBE_CONSUMER_0.as_mut().unwrap();
    common_handler(uart, cons);
}

#[interrupt]
unsafe fn UARTE1_UART1() {
    let uart = &*UARTE1::ptr();
    let cons: &mut Consumer = MAYBE_CONSUMER_1.as_mut().unwrap();
    common_handler(uart, cons);
}
```